OpenRA.Mods.Common/Lint/CheckDeathTypes.cs

```diff
@@ -36,6 +36,10 @@ public void Run(Action<string> emitError, Action<string> emitWarning, Ruleset ru
 				if (!targetable.Any())
 					continue;
 
+				var sa = actorInfo.Value.TraitInfos<SpawnActorOnDeathInfo>();
+				if (sa.Any())
+					deathTypes.AddRange(sa.Where(t => t.DeathType != null).Select(dt => dt.DeathType).ToList());
```

Comment:
This whole block can simplify down to something like:
`deathTypes.AddRange(actorInfo.Value.TraitInfos<SpawnActorOnDeathInfo>().SelectMany(s => s.DeathType).Where(dt => dt != null).Distinct()))`